index.js

```diff
@@ -0,0 +1,131 @@
+  });
+});
+
+// Currency converter
+function convertCurrency(event) {
+  event.preventDefault();
+  // Get the selected currencies and amount
+  const selectedFromCurrency = selectFromElement.value;
+  const selectedToCurrency = selectToElement.value;
+  const inputAmount = parseFloat(amount.value);
+
+  // Get the exchange rates for the selected currencies
+  const rateFrom = currencies.rates[selectedFromCurrency].rate;
+  const rateTo = currencies.rates[selectedToCurrency].rate;
+
+  // Perform the conversion and render it
+  const convertedAmount = (inputAmount / rateFrom) * rateTo;
+  const convertedAmountString = convertedAmount.toFixed(5);
+  const lastThreeDigits = convertedAmountString.slice(-3); // Get the last three digits
+  const selectedToCurrencyInfo = currencies.rates[selectedToCurrency];
+
+  amountWanted.innerHTML = `${inputAmount} ${selectedFromCurrency} =`;
+  amountResult.innerHTML = `${convertedAmountString.slice(
+    0,
+    -3
+  )}<span class="last-three-digits">${lastThreeDigits}</span> ${
+    selectedToCurrencyInfo.text
+  }`;
+  ratesNumbers.innerHTML = `1 ${selectedFromCurrency} = ${(
+    rateFrom / rateTo
+  ).toFixed(4)} ${selectedToCurrency}`;
+}
+
+// Add new currency function
+function addCurrency(event) {
+  event.preventDefault(); // Prevent form submission
+
+  const newRate = parseFloat(getRate.value);
+  const newCurrencyCode = getCode.value.toUpperCase();
+  const newCurrencyName = getName.value;
+
+  // Add the new currency to the currencies object
+  if (!currencies.rates[newCurrencyCode]) {
+    currencies.rates[newCurrencyCode] = {
+      rate: newRate,
+      text: newCurrencyName,
+    };
+    alert(
+      `New currency ${newCurrencyCode} - ${newCurrencyName} added with rate ${newRate}`
+    );
+
+    // Add the new currency to all dropdowns
+    const currencyDropdowns = document.querySelectorAll(".currency-dropdown");
+    currencyDropdowns.forEach((dropdown) => {
+      const newOption = document.createElement("option");
+      newOption.value = newCurrencyCode;
+      newOption.text = `${newCurrencyCode} - ${newCurrencyName}`;
+      dropdown.appendChild(newOption);
+    });
+  } else {
+    alert(`Currency code ${newCurrencyCode} already exists.`);
+  }
+  console.log(currencies); // for check if its added
+}
+
+// Update currency function
+function updateCurrency(event) {
+  event.preventDefault(); // Prevent form submission
+
+  const updatedRate = parseFloat(document.getElementById("update-rate").value);
+  const updatedCurrencyCode = document
+    .getElementById("to-currency-update")
+    .value.toUpperCase();
+
+  // Check if the currency code exists in the currencies object
+  if (currencies.rates[updatedCurrencyCode]) {
+    // Update the rate of the currency
+    currencies.rates[updatedCurrencyCode].rate = updatedRate;
+
+    alert(`Rate for currency ${updatedCurrencyCode} updated to ${updatedRate}`);
+  } else {
+    alert(`Currency code ${updatedCurrencyCode} does not exist.`);
```

Comment:
I'm not sure in which scenario you wouldn't be able to select an enabled currency, as they come from a dropdown. But I would say it's much better to have more validations than less. Well done. 👍

Reply:
no thats right, that is a logic mistake gonna fix it :D